Capstone Project

VicPeralta opened the pull request:
# Capstone project for React / Redux Module.

In this milestone:

- No linter errors.
- Correct GitFlow was used.
- Repo documented in a professional way.

## API
The API used was Published-Data API from Chess.com (Credit given on readme file)
## Design
The design followed was created by Nelson Sakwa (Credit given on readme file)
## Interactions
When the HomePage loads data is retrieved from the API and presented to the user, along with the average rating, calculated based on the data received.
When the user clicks on a category, the leaderboard for that category is displayed, if the user clicks on a player name, additional information is presented.
When the user clicks the back button, the user navigates to the HomePage or to the leaderboard, depending on where the user is.
## Testing
Tests were created for all components and reducer pure functions.
# Requeriments

- The project is a SPA built with React and Redux.
- The data retrieved is stored in the Redux Store.
- The data is filtered using a stateless component.
- Every page has a unique route within the SPA
- The project is deployed using GitHub pages and a link is provided in the readme file.
# Video
You can watch the presentation video [Here](https://www.loom.com/share/5623f03aa65740b4a9601ecc59c3bb77)

VicPeralta committed:
Create React App and set linters

new file:   .babelrc
new file:   .eslintrc.json
new file:   .github/workflows/linters.yml
new file:   .gitignore
new file:   .stylelintrc.json
new file:   LICENSE
modified:   README.md
new file:   README.old.md
new file:   package-lock.json
new file:   package.json
new file:   public/favicon.ico
new file:   public/index.html
new file:   public/logo192.png
new file:   public/logo512.png
new file:   public/manifest.json
new file:   public/robots.txt
new file:   src/App.css
new file:   src/App.js
new file:   src/App.test.js
new file:   src/index.css
new file:   src/index.js
new file:   src/logo.svg
new file:   src/reportWebVitals.js
new file:   src/setupTests.js

VicPeralta committed:
Merge pull request #1 from VicPeralta/setup

Setup React app and linters

VicPeralta committed:
Add initial general structure

Delete boilerplate
Update readme
install redux logger
install redux router
install redux-thunk
install react-redux
install prop-types
install react-icons
Configure store
Create basic actions
Create reducer
Create categoryCards
Create categoryList

modified:   README.md
deleted:    README.old.md
modified:   package-lock.json
modified:   package.json
modified:   public/index.html
modified:   src/App.css
modified:   src/App.js
new file:   src/Components/CategoryCard.css
new file:   src/Components/CategoryCard.js
new file:   src/Components/CategoryList.css
new file:   src/Components/CategoryList.js
new file:   src/Components/Fetching.js
new file:   src/Pages/HomePage.js
new file:   src/Redux/chessLeaders/chess.js
new file:   src/Redux/configureStore.js
new file:   src/assets/battle.png
new file:   src/assets/blitz.png
new file:   src/assets/blitz960.png
new file:   src/assets/bug.png
new file:   src/assets/bullet.png
new file:   src/assets/daily.png
new file:   src/assets/daily960.png
new file:   src/assets/house.png
new file:   src/assets/king.png
new file:   src/assets/ladybug.png
new file:   src/assets/rapid.png
new file:   src/assets/rush.png
new file:   src/assets/tactics.png
new file:   src/assets/three.png
modified:   src/index.js
deleted:    src/logo.svg
deleted:    src/reportWebVitals.js

VicPeralta committed:
Add routes and leaderboard component

Install react-router-dom

modified:   package-lock.json
modified:   package.json
modified:   src/App.css
modified:   src/App.js
modified:   src/Components/CategoryCard.css
modified:   src/Components/CategoryCard.js
new file:   src/Components/LeaderBoard.js
modified:   src/index.js

VicPeralta committed:
Add get player info to the reducer and corresponding actions

Create PlayerCard component
Add Routes to leaderboards and player cards paths

modified:   src/App.js
new file:   src/Components/PlayerCard.js
modified:   src/Redux/chessLeaders/chess.js

VicPeralta committed:
Add all routes and finish general structure

modified:   src/App.js
modified:   src/Components/CategoryCard.js
modified:   src/Components/LeaderBoard.js
modified:   src/Components/PlayerCard.js

VicPeralta committed:
Merge pull request #2 from VicPeralta/feature-general-structure

General structure

VicPeralta committed:
Add and style Homepage and fetching components

Add new icons for categories
Implemente search bar
Add credit for icons in readme file

modified:   README.md
modified:   src/App.css
modified:   src/Components/CategoryCard.css
modified:   src/Components/CategoryCard.js
modified:   src/Components/CategoryList.css
modified:   src/Components/CategoryList.js
new file:   src/Components/Fetching.css
modified:   src/Components/Fetching.js
new file:   src/Components/Header.css
new file:   src/Components/Header.js
new file:   src/Components/ProgressCircle.css
new file:   src/Components/ProgressCircle.js
modified:   src/Pages/HomePage.js
modified:   src/assets/battle.png
new file:   src/assets/battle_color.png
modified:   src/assets/blitz.png
modified:   src/assets/blitz960.png
new file:   src/assets/blitz960_color.png
new file:   src/assets/blitz_color.png
renamed:    src/assets/ladybug.png -> src/assets/bug_color.png
modified:   src/assets/bullet.png
new file:   src/assets/bullet_color.png
modified:   src/assets/daily.png
new file:   src/assets/daily_color.png
modified:   src/assets/house.png
new file:   src/assets/house_color.png
modified:   src/assets/king.png
new file:   src/assets/king_color.png
modified:   src/assets/rapid.png
new file:   src/assets/rapid_color.png
modified:   src/assets/rush.png
new file:   src/assets/rush_color.png
modified:   src/assets/tactics.png
new file:   src/assets/tactics_color.png
modified:   src/assets/three.png
new file:   src/assets/three_color.png
new file:   src/assets/whiteKing.png
new file:   src/assets/whiteQueen.png

VicPeralta committed:
Merge pull request #3 from VicPeralta/feature-home-fetch

HomePage & Fetching

VicPeralta committed:
Add SearchBar component

Modify CategoryList to use SearchBar component

modified:   src/App.css
modified:   src/Components/CategoryList.css
modified:   src/Components/CategoryList.js
new file:   src/Components/SearchBar.css
new file:   src/Components/SearchBar.js

VicPeralta committed:
Add PlayerCard and style Leaders and Players

modified:   src/Components/CategoryCard.css
modified:   src/Components/CategoryCard.js
new file:   src/Components/Leader.css
new file:   src/Components/Leader.js
new file:   src/Components/LeaderBoard.css
modified:   src/Components/LeaderBoard.js
new file:   src/Components/PlayerCard.css
modified:   src/Components/PlayerCard.js
modified:   src/Components/SearchBar.css
modified:   src/Components/SearchBar.js
new file:   src/assets/noPhoto.svg

VicPeralta committed:
Organize components into directories

modified:   src/App.js
renamed:    src/Components/CategoryCard.css -> src/Components/CategoryCard/CategoryCard.css
renamed:    src/Components/CategoryCard.js -> src/Components/CategoryCard/CategoryCard.js
renamed:    src/Components/CategoryList.css -> src/Components/CategoryList/CategoryList.css
renamed:    src/Components/CategoryList.js -> src/Components/CategoryList/CategoryList.js
renamed:    src/Components/Fetching.css -> src/Components/Fetching/Fetching.css
renamed:    src/Components/Fetching.js -> src/Components/Fetching/Fetching.js
renamed:    src/Components/Header.css -> src/Components/Header/Header.css
renamed:    src/Components/Header.js -> src/Components/Header/Header.js
renamed:    src/Components/Leader.css -> src/Components/Leader/Leader.css
renamed:    src/Components/Leader.js -> src/Components/Leader/Leader.js
renamed:    src/Components/LeaderBoard.css -> src/Components/LeaderBoard/LeaderBoard.css
renamed:    src/Components/LeaderBoard.js -> src/Components/LeaderBoard/LeaderBoard.js
renamed:    src/Components/PlayerCard.css -> src/Components/PlayerCard/PlayerCard.css
renamed:    src/Components/PlayerCard.js -> src/Components/PlayerCard/PlayerCard.js
renamed:    src/Components/ProgressCircle.css -> src/Components/ProgressCircle/ProgressCircle.css
renamed:    src/Components/ProgressCircle.js -> src/Components/ProgressCircle/ProgressCircle.js
renamed:    src/Components/SearchBar.css -> src/Components/SearchBar/SearchBar.css
renamed:    src/Components/SearchBar.js -> src/Components/SearchBar/SearchBar.js
modified:   src/Pages/HomePage.js

VicPeralta committed:
Merge pull request #4 from VicPeralta/feature-leaders

Feature leaders

VicPeralta committed:
Install gh-pages, update readme file

renamed:    LICENSE -> LICENSE.md
modified:   README.md
new file:   chessLeaders.png
new file:   chessLeadersBlack.png
modified:   package-lock.json
modified:   package.json

VicPeralta committed:
Merge pull request #5 from VicPeralta/gh-pages-deploy

GH-Pages

VicPeralta committed:
Add tests for some components

Update react testing library from 12.1 to 13.01
CategoryList, Fetching, Header, ProgressCircle, SearchBar
install react-test-renderer

modified:   package-lock.json
modified:   package.json
modified:   src/App.js
modified:   src/App.test.js
modified:   src/Components/CategoryList/CategoryList.js
new file:   src/Components/CategoryList/CategoryList.test.js
new file:   src/Components/CategoryList/__snapshots__/CategoryList.test.js.snap
new file:   src/Components/Fetching/Fetching.test.js
new file:   src/Components/Header/Header.test.js
modified:   src/Components/ProgressCircle/ProgressCircle.js
new file:   src/Components/ProgressCircle/ProgressCircle.test.js
modified:   src/Components/SearchBar/SearchBar.js
new file:   src/Components/SearchBar/SearchBar.test.js

VicPeralta committed:
Add test for all components and reducer

modified:   package.json
new file:   src/Components/CategoryCard/CategoryCard.test.js
new file:   src/Components/CategoryCard/__snapshots__/CategoryCard.test.js.snap
modified:   src/Components/CategoryList/CategoryList.test.js
modified:   src/Components/CategoryList/__snapshots__/CategoryList.test.js.snap
modified:   src/Components/Fetching/Fetching.test.js
new file:   src/Components/Fetching/__snapshots__/Fetching.test.js.snap
modified:   src/Components/Header/Header.test.js
new file:   src/Components/Header/__snapshots__/Header.test.js.snap
new file:   src/Components/Leader/Leader.test.js
new file:   src/Components/Leader/__snapshots__/Leader.test.js.snap
new file:   src/Components/LeaderBoard/LeaderBoard.test.js
new file:   src/Components/LeaderBoard/__snapshots__/LeaderBoard.test.js.snap
modified:   src/Components/PlayerCard/PlayerCard.js
new file:   src/Components/PlayerCard/PlayerCard.test.js
new file:   src/Components/PlayerCard/__snapshots__/PlayerCard.test.js.snap
modified:   src/Components/SearchBar/SearchBar.test.js
new file:   src/Components/SearchBar/__snapshots__/SearchBar.test.js.snap
new file:   src/Redux/chessLeaders/chess.test.js

VicPeralta committed:
Merge pull request #6 from VicPeralta/feature-testing

Feature testing

VicPeralta committed:
Add metadata to index.html and updates readme

modified:   README.md
modified:   public/index.html

VicPeralta committed:
Merge pull request #7 from VicPeralta/metadata

Add metadata to index.html and updates readme

Deepakdanger commented on README.md:
```diff
@@ -1,2 +1,60 @@
+## Getting Started
+
+To get a local copy follow these simple example steps.  
+
+1.- Open the terminal window and clone the repository using this command:  
+`git@github.com:VicPeralta/chess-leaders.git` 
+
+2.- Change the directory to the react-bookstore directory  
+`cd chess-leaders`  
+3.- Install the project's dependencies by running this command:   
+`npm install`  
+4.- Generate the dist folder using this command:  
+`npm run build`  
+5.- Run this command to see the page `npm start`  
+6.- To see the code open the project in the code editor of your preference.  
+
```
- [x] I would recommend adding the command to test the application. `npm test`.

Deepakdanger reviewed (changes requested):
Hi @VicPeralta, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Highlights
:+1: The Application is deployed online.
:+1: The Application follows the design.

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

VicPeralta committed:
Add change suggested by the reviewer

Fix some style issues

modified:   README.md
modified:   src/App.css
modified:   src/Components/CategoryCard/CategoryCard.css
modified:   src/Components/LeaderBoard/LeaderBoard.js
modified:   src/Components/LeaderBoard/LeaderBoard.test.js
modified:   src/Components/LeaderBoard/__snapshots__/LeaderBoard.test.js.snap
modified:   src/Components/SearchBar/SearchBar.js

VicPeralta committed:
Fix error

VicPeralta committed:
Merge pull request #9 from VicPeralta/code-review

Add change suggested by the reviewer

Whoistolu reviewed (approved):
### Hi @VicPeralta

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

- [ ] [OPTIONAL] You have done a great project, however, kindly note that it is very important to always highlight a piece of the code you are proud of making in the presentation video.
- [ ] Also, it will be cooler to include a link to the presentation video in the README of the project instead of attaching it to just the PR. The README tells a lot of stories about the project to users/visitors.

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
